docs/cli-preview3/tools/index.md

```diff
@@ -0,0 +1,129 @@
+The following example utilizes several commands from the CLI standard install to initialize a new simple console application, 
+restore the dependencies, build the application and then run it. 
+
+```console
+dotnet new
+dotnet restore
+dotnet build --output /stuff
+dotnet /stuff/new.dll
+```
+
+As you can see in the previous example, there is a pattern in the way you use the CLI tools. Within that pattern, we can 
+identify three main pieces of each command:
+
+1. [The driver ("dotnet")](#driver)
+2. [The command, or "verb"](#the-verb)
+3. [Command arguments](#the-arguments)
+
+### Driver
+The driver is named [dotnet](dotnet.md). It is the first part of what you invoke. The driver has two responsibilities:
+
+1. Running portable apps
+2. Executing the verb
+
+What it does depends on what is specified on the command line. In the first case, you would 
+specify a portable app DLL that `dotnet` would run similar to this: `dotnet /path/to/your.dll`. 
+
+In the second case, the driver attempts to invoke the specified command. This starts the CLI command execution 
+process. First, the driver determines the version of the tooling that you want. You can specify the version in the 
+[global.json](global-json.md) file using the `version` property. If that is not available, the driver finds the latest version
+of the tools that is installed on disk and uses that version. Once the version is determined, it executes the 
+command. 
+
+### The "verb"
+The verb is simply a command that performs an action. `dotnet build` builds your code. `dotnet publish` publishes 
+your code. The verb is implemented as a console application that is named per convention: `dotnet-{verb}`. All of the 
+logic is implemented in the console application that represents the verb. 
+
+### The arguments
+The arguments that you pass on the command-line are the arguments to the actual verb/command being invoked. 
+For example, when you type `dotnet publish --output publishedapp`, the `--output` argument is passed to the 
+`publish` command. 
+
+## Types of application portability
+CLI enables applications to be portable in two main ways:
+
+1. Completely portable applications that can run anywhere .NET Core is installed
+2. Self-contained deployments
+
+You can learn more about both of these in the [.NET Core application deployment](../deploying/index.md) topic. 
+
+## Migration from DNX
+If you used DNX in .NET Core 1.0 RC1, you may be wondering what happened to it and how do these new tools
+relate to the DNX tools. In short, the DNX tools have been replaced with the .NET Core CLI tools. 
+If you have existing projects or are just wondering how the commands map, you
+can use the [DNX to CLI migration](../migrating-from-dnx.md) topic to get all the details. 
+
+## Extensibility
+Of course, not every tool that you could use in your workflow will be part of the core CLI tools. However, .NET Core 
```

Comment:
suggestion: I'd reverse the clauses of the first sentence to sound stronger:

"The .NET CLI provides a set of verbs that handles many scenarios. It has an extenibility ... for your projects for scenarios that are not part of the CLI install."